app/bundles/CoreBundle/Helper/BuilderTokenHelper.php

```diff
@@ -11,37 +11,57 @@
 
 namespace Mautic\CoreBundle\Helper;
 
+use Doctrine\DBAL\Connection;
 use Doctrine\DBAL\Query\Expression\CompositeExpression;
-use Mautic\CoreBundle\Factory\MauticFactory;
+use Doctrine\ORM\EntityManager;
+use Mautic\CoreBundle\Security\Permissions\CorePermissions;
 
 /**
  * Class BuilderTokenHelper.
  */
 class BuilderTokenHelper
 {
-    protected $viewPermissionBase;
-    protected $modelName;
-    protected $langVar;
-    protected $bundleName;
+    private $isConfigured = false;
+
+    private $security;
+    private $entityManager;
+    private $connection;
+    private $userHelper;
+
     protected $permissionSet;
+    protected $modelName;
+    protected $viewPermissionBase = null;
+    protected $langVar            = null;
+    protected $bundleName         = null;
 
     /**
-     * @var MauticFactory
-     *
-     * @deprecated 2.12 To be removed in 3.0 Inject dependencies in your constructor instead
+     * @param CorePermissions $security
+     * @param EntityManager   $entityManager
+     * @param Connection      $connection
+     * @param UserHelper      $userHelper
      */
-    private $factory;
+    public function __construct(
+        CorePermissions $security,
+        EntityManager $entityManager,
+        Connection $connection,
+        UserHelper $userHelper
+    ) {
+        $this->security      = $security;
+        $this->entityManager = $entityManager;
+        $this->connection    = $connection;
+        $this->userHelper    = $userHelper;
+    }
 
     /**
-     * @param MauticFactory $factory
-     * @param string        $modelName          Model name such as page
-     * @param string        $viewPermissionBase Permission base such as page:pages or null to generate from $modelName
-     * @param string        $bundleName         Bundle name such as MauticPageBundle or null to generate from $modelName
-     * @param null          $langVar            Language base for filter such as page.page or leave blank to use $modelName
+     * This method must be called before the BuilderTokenHelper can be used.
+     *
+     * @param $modelName
+     * @param null $viewPermissionBase
+     * @param null $bundleName
+     * @param null $langVar
```

Comment:
```suggestion
     * @param string|null $langVar
```
The other params are missing types as well